libstorage/storage.c

```diff
@@ -333,49 +409,50 @@ static int storagectx_init(request_ctx_t *ctx, handler_t handler, void *fs)
 
 storage_t *storage_get(int id)
 {
-	return lib_treeof(storage_t, node, idtree_find(&storage_common.devs, id));
+	return lib_treeof(storage_t, node, idtree_find(&storage_common.strgs, id));
 }
 
 
-static filesystem_t *storage_getfs(const char *name)
+static storage_fsHandler_t *storage_getfs(const char *name)
 {
-	filesystem_t fs;
+	storage_fsHandler_t fs;
 
 	strncpy(fs.name, name, sizeof(fs.name));
 	fs.name[sizeof(fs.name) - 1] = '\0';
 
-	return lib_treeof(filesystem_t, node, lib_rbFind(&storage_common.fss, &fs.node));
+	return lib_treeof(storage_fsHandler_t, node, lib_rbFind(&storage_common.fss, &fs.node));
 }
 
 
-int storage_registerfs(const char *name, mount_t mount, umount_t umount, handler_t handler)
+int storage_registerfs(const char *name, mount_t mount, umount_t umount)
 {
-	filesystem_t *fs;
+	storage_fsHandler_t *handler;
 
-	if ((name == NULL) || (mount == NULL) || (umount == NULL) || (handler == NULL))
+	if ((name == NULL) || (mount == NULL) || (umount == NULL))
 		return -EINVAL;
 
-	if ((fs = malloc(sizeof(filesystem_t))) == NULL)
+	if ((handler = malloc(sizeof(storage_fsHandler_t))) == NULL)
```

Comment:
Sometimes you avoid assignment in if, sometimes not. Let's stick to avoiding

Reply:
Done